File: w12/d00 Testing & TDD/wizard/lib/wizard.rb
class Wizard
  attr_reader :magic
  attr_reader :health

  def initialize
    @magic = 5
    @health = 2
  end

  def cast_spell
    if @magic > 0 && @health > 0
      @magic -= 1
      true
    else
      false
    end
  end

  def regenerate_magic
    @magic = 5
  end

  def regenerate_health
    @health = 2
  end

  def hit
    if @health > 0
      @health -= 1
      true
    else
      false
    end
  end
end

File: w12/d00 Testing/wizard/lib/wizard.js
var Wizard = function() {
  this.magicLevel = 5;
  this.health = 2;

  this.castSpell = function() {
    if(this.magicLevel > 0 && this.health > 0) {
      this.magicLevel -= 1;
      return true;
    }
    else {
      return false;
    }
  };

  this.regenSpell = function() {
    this.magicLevel = 5;
  };

  this.regenHealth = function() {
    this.health = 2;
  };

  this.hit = function() {
    if(this.health > 0) {
      this.health -= 1;
      return true;
    }
    else {
      return false;
    }
  };
};

module.exports = Wizard;
